framework/src/multiapps/MultiApp.C

```diff
@@ -661,6 +670,18 @@ MultiApp::createApp(unsigned int i, Real start_time)
   app->setupOptions();
   preRunInputFile();
   app->runInputFile();
+
+  FEProblemBase & fe_problem_base = _apps[i]->getExecutioner()->feProblem();
+  auto & solver_params = fe_problem_base.solverParams();
+  solver_params._picard_self_relaxation_factor = getParam<Real>("relaxation_factor");
+  solver_params._picard_self_relaxed_variables =
+      getParam<std::vector<std::string>>("relaxed_variables");
+  if (solver_params._picard_self_relaxation_factor != 1.0)
+  {
+    // Store a copy of the previous solution here
+    fe_problem_base.getNonlinearSystemBase().addVector("self_relax_previous", false, PARALLEL);
+    fe_problem_base.previousTimeAsSubapp() = fe_problem_base.time() - 1;
+  }
```

Comment:
This seems weird - it shouldn't go here me thinks.

Reply:
This is the trick used in the old implementation. It is used to check if we are re-entering `newSolve` in the same time step, which can happen with Picard iteration on master with subapps.